src/Illuminate/Routing/Router.php

```diff
@@ -360,6 +361,21 @@ public function resource($name, $controller, array $options = [])
     }
 
     /**
+     * Register a new redirect route with the router.
+     *
+     * @param  string  $from
+     * @param  string  $to
+     * @param  int  $status
+     * @return \Illuminate\Routing\Route
+     */
+    public function redirect($from, $to, $status = 301)
+    {
+        return $this->get($from, function () use ($to, $status) {
+            return new RedirectResponse($to, $status);
+        });
```

Comment:
Difference is, the cacher should be able to handle this by-default. So a method should be found to allow for redirects to be cached by default. Perhaps an internal controller that has the redirect method and signature?

Redirect should not come with the warning that in order to cache you handle it manually, that makes it not nearly as useful as it would otherwise be.